src/providers/consensus/client.py

```diff
@@ -33,9 +34,33 @@ class ConsensusClient(HTTPProvider):
     API_GET_BLOCK_HEADER = '/eth/v1/beacon/headers/{}'
     API_GET_BLOCK_DETAILS = 'eth/v2/beacon/blocks/{}'
     API_GET_VALIDATORS = 'eth/v1/beacon/states/{}/validators'
+    _API_GET_CONSENSUS_SPEC = 'eth/v1/config/spec'
 
     NON_CACHEABLE_STATES = ('head', 'finalized', 'justified')
 
+    spec: ConsensusSpecRequiredOnly
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.spec = self._get_consensus_spec()
+
+    def renew_spec(self):
+        self.spec = self._get_consensus_spec()
+
+    def _get_consensus_spec(self) -> ConsensusSpecRequiredOnly:
+        """
+        Spec: https://ethereum.github.io/beacon-APIs/#/Config/getSpec
+
+        We don't use response dataclass init 'as-is' here because new fields may appear and brake struct.
+        We should only take what we need
+
+        No cache because this method is using without any args
+        """
+        data, _ = self._get(self._API_GET_CONSENSUS_SPEC)
+        return ConsensusSpecRequiredOnly(
+            **{k: v for k, v in data.items() if k in ConsensusSpecRequiredOnly.__annotations__}
+        )
```

Comment:
- can we cover to `int` here?
- what will happen if one of the required keys does not exist?

Reply:
1. not all values are numbers. not sure that we need implement some convert logic by types in dataclass
2. exception when creating dataclass object